src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -79,9 +88,41 @@ export default class Pools {
     } catch {
       // eslint-disable-previous-line no-empty
     }
+
+    if (this.excludedAddresses == null) {
+      // fetch only once
+      this.excludedAddresses = await this.getExcludedAddresses();
+    }
+
     return this.serialize(pools, pastPools, period, prices, currency);
   }
 
+  public removeExcludedAddressesFromTotalLiquidity(
```

Comment:
was just wondering if for future whether we should be keeping methods like this inside the service or have more as a isolated helper function?

Reply:
Yeah prob could be a good idea - I think we should keep this in mind as part of a larger refactor we want to do around the pools stuff.

Reply:
Yea +1 to making it a generic `removeAddressesFromTotalLiquidity` and the excluded addresses is passed as the first argument.